optimize/backend/src/test/java/io/camunda/optimize/service/util/BpmnModelUtilTest.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.optimize.service.util;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import io.camunda.optimize.dto.optimize.FlowNodeDataDto;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import org.camunda.bpm.model.bpmn.BpmnModelInstance;
+import org.junit.jupiter.api.Test;
+
+public class BpmnModelUtilTest {
+
+  private static final String SIMPLE_BPMN_RESOURCE_PATH = "/test/data/simpleBpmn.xml";
+
+  private String getBpmnXml() throws IOException {
+    return BpmnModelUtil.getResourceFileAsString(SIMPLE_BPMN_RESOURCE_PATH);
+  }
+
+  @Test
+  void testParseBpmnModel() throws IOException {
+    final BpmnModelInstance modelInstance = BpmnModelUtil.parseBpmnModel(getBpmnXml());
+    assertNotNull(modelInstance, "BPMN model should be parsed correctly.");
+  }
+
+  @Test
+  void testExtractFlowNodeData() throws IOException {
+    final List<FlowNodeDataDto> flowNodeData = BpmnModelUtil.extractFlowNodeData(getBpmnXml());
+    assertNotNull(flowNodeData, "FlowNodeData list should not be null.");
+    assertFalse(flowNodeData.isEmpty(), "FlowNodeData list should not be empty.");
+
+    final FlowNodeDataDto firstNode = flowNodeData.get(0);
```

Comment:
with a simpler model (see comment at the beginning of this class), we can even assert on all of the nodes rather than just the first one

Reply:
fixed